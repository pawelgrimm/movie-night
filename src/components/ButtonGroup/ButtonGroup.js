import React from "react";
import classNames from "classnames";

const ButtonGroup = ({ children, className }) => {
  return (
    <div className={classNames("button-group", className)}>{children}</div>
  );
};

export default ButtonGroup;
