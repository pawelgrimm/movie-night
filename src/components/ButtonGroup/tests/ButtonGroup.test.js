import { expect, jest } from "@jest/globals";
import React from "react";
import { shallow } from "enzyme";
import ButtonGroup from "../ButtonGroup";

test("should render button group", () => {
  const wrapper = shallow(
    <ButtonGroup>
      <button>Button Text</button>
      <button>Button Text</button>
    </ButtonGroup>
  );
  expect(wrapper).toMatchSnapshot();
});

test("should have correct class", () => {
  const wrapper = shallow(<ButtonGroup />);
  expect(wrapper.hasClass("button-group")).toBeTruthy();
});

test("should allow additional classes", () => {
  const wrapper = shallow(<ButtonGroup className="some-class" />);
  expect(wrapper.hasClass("button-group")).toBeTruthy();
  expect(wrapper.hasClass("some-class")).toBeTruthy();
});
