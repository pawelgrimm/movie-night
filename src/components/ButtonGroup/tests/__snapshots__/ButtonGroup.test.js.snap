// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render button group 1`] = `
<div
  className="button-group"
>
  <button>
    Button Text
  </button>
  <button>
    Button Text
  </button>
</div>
`;
